tutorials/data-science-extraction/index.md

```diff
@@ -0,0 +1,390 @@
+            include_syntax=True, include_entities=False,
+            include_sentiment=False)
+
+    return tokens, metadata
+```
+
+We can write a short `main` function to confirm it works:
+
+[embedmd]:# (src/extraction/sentence_structure.py /def main/ /print.*/)
+```py
+def main(input_files):
+    for line in fileinput.input(input_files):
+        logging.info('Analyzing "{}"'.format(line))
+
+        tokens, metadata = extract_syntax([line], {})
+        for token in tokens:
+            print('{}: {}'.format(token.text_content, token.part_of_speech))
+```
+
+Running this on a sample phrase produces:
+
+    $ python sentence_structure.py - <<EOF
+    Everyone shall sit under their own vine and fig tree, and no one shall make them afraid.
+    EOF
+
+    Everyone: NOUN
+    shall: VERB
+    sit: VERB
+    under: ADP
+    their: PRON
+    own: ADJ
+    vine: NOUN
+    and: CONJ
+    fig: NOUN
+    tree: NOUN
+    ,: PUNCT
+    and: CONJ
+    no: DET
+    one: NOUN
+    shall: VERB
+    make: VERB
+    them: PRON
+    afraid: ADJ
+    .: PUNCT
+
+We've now gone from a stream of spoken prose, transformed it into a form
+readable by our tools, and came out with a structured catalog of its contents.
+In this form, we can unleash our exploratory tools, as described in [Exploration
+using queries](bigquery.md).
+
+[comment]: # (and [Visualization with interactive notebooks].)
+
+But first, it's imperative that we go from manually transforming this data with
+a series of scripts, to [automating this process](preprocessing.md).
+
+## API Documentation & other resources
+
+We've only touched on a couple of the capabilities of the APIs we've used here.
+Take a look at the API documentation, and experiment with the other features.
+
+* Speech API ([API Docs][speech-docs] | [Client Library Docs][speech-client])
+* Natural Language API ([API Docs][natural-language-docs] | [Client Library Docs][language-client])
+* Vision API ([API Docs][vision-docs] | [Client Library Docs][vision-client])
+* Translate API ([API Docs][translate-docs] | [Client Library Docs][translate-client])
+
+Also, take a look at the [Structuring Unstructured text][unstructured-text] demo
+for another example of using the Natural Language API.
+
+[speech-docs]: /speech/docs
+[natural-language-docs]: /natural-language/docs
+[vision-docs]: /vision/docs
+[translate-docs]: /translate/docs
+[speech-client]: //googlecloudplatform.github.io/google-cloud-python/stable/speech-usage.html
+[language-client]: //googlecloudplatform.github.io/google-cloud-python/stable/language-usage.html
+[vision-client]: //googlecloudplatform.github.io/google-cloud-python/stable/vision-usage.html
+[translate-client]: //googlecloudplatform.github.io/google-cloud-python/stable/translate-usage.html
+[unstructured-text]: /blog/big-data/2016/08/structuring-unstructured-text-with-the-google-cloud-natural-language-api
+
+## What's next
+
+The astute among you may notice that the return values of each of these steps
```

Comment:
Mm... I'd prefer to leave it. It's more stylistic than anything, and while I acknowledge that there are instances where a reader can reasonably infer an insult where one wasn't necessarily meant (and one should try to avoid that), I'd like to avoid making my writing bland for fear of hypothetical sensitive souls :-)

For the record, I didn't notice the `0.0` values until after I'd finished this article and was halfway through the exploration article ^_^